### PR TITLE
I97 - fix saving/loading

### PR DESCRIPTION
**Pull request by hillalex**
But first merge the other outstanding PRs, as this contains commits from both of them.

This PR also fixes i83 - don't allow pjnz files for multiple countries.

### CODE REVIEW

File: src/server/survey_and_program_data.R [hunk 1]
```diff
@@ -1,58 +1,83 @@
...
-    data(survey_hts)
-    data(prgm_dat)
+    data("survey_hts", package="first90")
+    data("prgm_dat", package="first90")
+
+    spectrumFilesState$newCountry <- FALSE
+
+    shiny::observeEvent(spectrumFilesState$newCountry, {
 
-    observeEvent(spectrumFilesState$country, {
```
Comment (hillalex) [inline src/server/survey_and_program_data.R]: This was firing after the digest was read in, so over-writing the loaded survey and program data. Now listening for changes to `spectrumFileState$newCountry` instead, which only gets written to on upload of the first pjnz file.

File: src/server/spectrum_files.R [hunk 2]
```diff
@@ -11,26 +12,40 @@ spectrumFiles <- function(input, output, state) {
...
+
+                dataSet = list(name = inFile$name,
+                                data = first90::prepare_inputs(inFile$datapath))
+
+                state$dataSets <- c(state$dataSets, list(dataSet))
+            }
+            else if (state$country != newCountry){
+                state$multipleCountryError <- TRUE
+                shinyjs::reset("spectrumFile")
+            }
```
Comment (MartinEden) [inline src/server/spectrum_files.R:37]: I don't think this looks right. In the case where you upload a second file of the same country, we're now no longer doing anything with it, right? Whereas we should be adding it to `state$dataSets`.

File: src/server/spectrum_files.R [hunk 3]
```diff
@@ -11,26 +12,40 @@ spectrumFiles <- function(input, output, state) {
     })
 
     shiny::observeEvent(input$spectrumFile, {
+        state$newCountry <- FALSE
         inFile <- input$spectrumFile
+        state$multipleCountryError <- FALSE
+        state$newCountry <- FALSE
```
Comment (MartinEden) [inline src/server/spectrum_files.R]: You defined this to be FALSE twice

**Review by MartinEden — CHANGES_REQUESTED** [review]

File: src/server/spectrum_files.R (hunk 2 above)
Comment (hillalex) [reply src/server/spectrum_files.R:37]: Yes you're right

File: src/server/spectrum_files.R (hunk 3 above)
Comment (hillalex) [reply src/server/spectrum_files.R]: Removed the first definition

**Comment by MartinEden** [comment]
I think this is still a little confusing, as you have the local variable `newCountry`, which is the name of the country just read from the spectrum file. And then you have the state level `state$newCountry`, which is a boolean, and tracks whether a country has been added ever? I'm also not sure that this second variable is needed. The only place it is used I think is

```
spectrumFilesState$newCountry && !is.null(spectrumFilesState$country)
```

but I don't think one side of this expression will ever be true without the other one, so it's redundant?

**Comment by hillalex** [comment]
You're right, I cant see why I thought this was needed.

**Review by MartinEden — APPROVED** [review]

**Pull request merged**